lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -0,0 +1,59 @@
+"""
+Platform related operations for Mobile APP
+"""
+import abc
+import re
+
+
+class MobilePlatform:
+    """
+    MobilePlatform class creates an instance of platform based on user agent and supports platform
+    related operations
+    """
+    __metaclass__ = abc.ABCMeta
+    version = None
+
+    def __init__(self, version):
+        self.version = version
+
+    @classmethod
+    def create_instance(cls, user_agent, user_agent_regex):
```

Comment:
Nit: rename these to simply `create` and `get` to be consistent with other such factory names in the system.